### PR TITLE
Rephrase exception message from CORS add

### PR DESCRIPTION
**Pull request by troydai**
Issue #3362 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/services_wrapper.py [hunk 1]
```diff
@@ -41,7 +42,15 @@ def add_cors(self, origins, methods, max_age, exposed_headers=None, allowed_head
         cors = self.get_cors(timeout)
         new_rule = CorsRule(origins, methods, max_age, exposed_headers, allowed_headers)
         cors.append(new_rule)
-        return self.set_service_properties()(cors=cors, timeout=timeout)
+        try:
+            return self.set_service_properties()(cors=cors, timeout=timeout)
+        except AzureHttpError as ex:
+            import re
+            message = re.search(r"([A-Za-z\t .])+", str(ex))
+            if message.group(0) == 'XML specified is not syntactically valid.':
```
Comment (tjprescott) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/services_wrapper.py]: Is this a service workaround?  If so, we should create a tracking issue so we can remove this when (!) the service fixes its ways.

Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/services_wrapper.py]: Will change.

**Comment by troydai** [comment]
@tjprescott I change the solution,

**Comment by tjprescott** [comment]
@troydai I don't really have an issue with message.  It looks like this is a case where the server is returning a junk error message, just like here: https://github.com/Azure/azure-storage-python/issues/181

I'm fine with the workaround so our customers don't suffer, but we should create a tracking issue either in our repo with the ServiceAttn label or in the Python SDK repo (where they will give it the ServerIssue label) and comment the workaround with this issue so that when it is resolved, we can remove it.

Right now, if the service started supporting more than 5 CORS rules, we're making bad assumptions.

**Comment by troydai** [comment]
There is an issue filed on the service side. They're going to fix the error message.

**Comment by troydai** [comment]
ping @tjprescott

**Comment by tjprescott** [comment]
Right, that's what I mean. There should be a comment in the code that says something to the effect of "TODO: Remove once issue python-storage-sdk/1234 is resolved".  Otherwise this logic will never be removed.

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/services_wrapper.py [hunk 2]
```diff
@@ -41,7 +42,13 @@ def add_cors(self, origins, methods, max_age, exposed_headers=None, allowed_head
         cors = self.get_cors(timeout)
         new_rule = CorsRule(origins, methods, max_age, exposed_headers, allowed_headers)
         cors.append(new_rule)
-        return self.set_service_properties()(cors=cors, timeout=timeout)
+        try:
+            return self.set_service_properties()(cors=cors, timeout=timeout)
+        except AzureHttpError as ex:
+            if ex.status_code == 400 and len(cors) > 5:
+                raise CLIError('Failed to add CORS rule. No more than 5 CORS rule can be added.')
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/services_wrapper.py]: I might be wrong, but should we use plurals like `rules`?

**Comment by troydai** [comment]
@tjprescott @yugangw-msft updated.

**Pull request merged**